### DISCUSSION TITLE
Python bindings build error due to submodules

### DISCUSSION THREAD

**Discussion by kam56** [bitcraze/discussions#633] [Q&A]
I get the following error on mac while trying to clone the crazyflie-firmware repo with --recursive tag:

`Unable to checkout 'a65b7c9a3e6502127fdb80eb288d8cbdf251a6f4' in submodule path 'vendor/CMSIS'`

Which causes the following error when I generate python bindings:

```
clang: error: no such file or directory: 'vendor/CMSIS/CMSIS/DSP/Source/BasicMathFunctions/arm_add_f32.c'
clang: error: no input files
error: command '/usr/bin/clang' failed with exit code 1
```

Any idea on how to solve this issue?

**Comment by knmcguire** [comment]
Hi!

Indeed this is due to the submodule not being downloaded so good catch! I've changed the title a bit.

I've tried to reclone the repo on an Ubuntu machine, but I'm not seeing this issue of not being able to checkout the repo. Have you tried removing the repository and trying it again?

What you could also try in the existing repo, is the following
```
git submodule init
git submodule sync
git submodule update
```

If that doesn't work, then it is probably out of our scope. As this seems to be a generic git problem so stackoverflow/google should have an answer somewhere.

**Reply by kam56** [reply]
Hi, 

Thank you for the reply. I tried it on Ubuntu and it works. 
But it still does not work on Mac. Unfortunately, I couldn't figure out what causes the error. 

`Unable to checkout 'a65b7c9a3e6502127fdb80eb288d8cbdf251a6f4' in submodule path 'vendor/CMSIS'`

But thank you anyways :)

**Reply by krichardsson** [reply]
> `Unable to checkout 'a65b7c9a3e6502127fdb80eb288d8cbdf251a6f4' in submodule path 'vendor/CMSIS'` 

This sounds more like a git problem than a mac problem. Maybe you could try to clone the repo recursively from the start, `git clone --recursive ....`
